transit_odp/timetables/dataframes.py

```diff
@@ -256,32 +258,75 @@ def vehicle_journeys_to_dataframe(
     return pd.DataFrame(all_vechicle_journeys)
 
 
-def operating_profile_to_df(operating_profiles):
-    serviced_orgs_refs_df = []
-    for operating_profile in operating_profiles:
-        serviced_organisation_day_type = operating_profile.get_element_or_none(
-            ["ServicedOrganisationDayType"]
+def populate_operating_profiles(operating_profile_services):
+    serviced_org_ref = ""
+    days_of_week = ""
+    operational = ""
+    serviced_organisation_day_type = operating_profile_services.get_element_or_none(
+        ["ServicedOrganisationDayType"]
+    )
+    regular_day_type = operating_profile_services.get_element_or_none(
+        ["RegularDayType"]
+    )
+    if regular_day_type:
+        days_of_week_elements = regular_day_type.get_elements_or_none(["DaysOfWeek"])
+        if days_of_week_elements:
+            days_of_week_element = days_of_week_elements[0]
+            days_of_week = [day.localname for day in days_of_week_element.children]
+    if serviced_organisation_day_type:
+        days_of_operation = serviced_organisation_day_type.get_element_or_none(
+            ["DaysOfOperation"]
         )
-        if serviced_organisation_day_type:
-            days_of_operation = serviced_organisation_day_type.get_element_or_none(
-                ["DaysOfOperation"]
-            )
-            days_of_non_operation = serviced_organisation_day_type.get_element_or_none(
-                ["DaysOfNonOperation"]
+        days_of_non_operation = serviced_organisation_day_type.get_element_or_none(
+            ["DaysOfNonOperation"]
+        )
+        if days_of_operation:
+            operational = True
+            working_days = days_of_operation.get_element("WorkingDays")
+        elif days_of_non_operation:
+            operational = False
+            working_days = days_of_non_operation.get_element("WorkingDays")
+        serviced_org_ref = working_days.get_element("ServicedOrganisationRef").text
+
+    return serviced_org_ref, days_of_week, operational
+
+
+def operating_profiles_to_dataframe(vehicle_journeys, services):
+    operating_profile_list = []
+    for vehicle_journey in vehicle_journeys:
+        vehicle_journey_code = vehicle_journey.get_element(["VehicleJourneyCode"]).text
+        service_ref = vehicle_journey.get_element(["ServiceRef"]).text
+        operating_profile_vehicle_journey = vehicle_journey.get_element_or_none(
+            ["OperatingProfile"]
+        )
+        if operating_profile_vehicle_journey:
+            operating_profile = operating_profile_vehicle_journey
+        else:
+            for service in services:
```

Comment:
In case there are more than one registered services with operating profile in them - this would not process the data as expected. It would only process the last registered service operating profile